test/e2e/framework/test.go

```diff
@@ -100,10 +107,14 @@ func RunParallel(top *testing.T, name string, f TestFunc, cfgs ...KcpConfig) {
 			runningServers[server.name] = server
 		}
 
+		// Use a context to communicate when the test-executing
+		// goroutine has finished.
+		testCompleteCtx, testCompleteCancel := context.WithCancel(context.Background())
```

Comment:
Do you want the parent to be ctx from L84 or for this to be disassociated from that?